File: java/Ejercicio_10.java
import java.util.HashMap;
import java.util.Map;
import java.util.Scanner;

public class Ejercicio_10{
    public static void main(String[] args) {
        Map<String, Map<String, String>> clientes = new HashMap<>();
        Scanner scanner = new Scanner(System.in);

        while (true) {
            // Mostrar el menú de opciones
            System.out.println("\nSeleccione una opción:");
            System.out.println("1. Añadir cliente");
            System.out.println("2. Eliminar cliente");
            System.out.println("3. Mostrar cliente");
            System.out.println("4. Listar todos los clientes");
            System.out.println("5. Listar clientes preferentes");
            System.out.println("6. Terminar");
            System.out.print("Ingrese su opción: ");
            String opcion = scanner.nextLine();

            switch (opcion) {
                case "1":
                    // Añadir un nuevo cliente
                    System.out.print("Ingrese el NIF del cliente: ");
                    String nif = scanner.nextLine();
                    System.out.print("Ingrese el nombre del cliente: ");
                    String nombre = scanner.nextLine();
                    System.out.print("Ingrese la dirección del cliente: ");
                    String direccion = scanner.nextLine();
                    System.out.print("Ingrese el teléfono del cliente: ");
                    String telefono = scanner.nextLine();
                    System.out.print("Ingrese el correo electrónico del cliente: ");
                    String correo = scanner.nextLine();
                    System.out.print("¿Es un cliente preferente? (sí/no): ");
                    String preferente = scanner.nextLine().equalsIgnoreCase("sí") ? "sí" : "no";

                    Map<String, String> cliente = new HashMap<>();
                    cliente.put("Nombre", nombre);
                    cliente.put("Dirección", direccion);
                    cliente.put("Teléfono", telefono);
                    cliente.put("Correo", correo);
                    cliente.put("Preferente", preferente);

                    clientes.put(nif, cliente);
                    System.out.println("Cliente " + nombre + " añadido con éxito.");
                    break;

                case "2":
                    // Eliminar un cliente existente
                    System.out.print("Ingrese el NIF del cliente a eliminar: ");
                    nif = scanner.nextLine();
                    if (clientes.containsKey(nif)) {
                        clientes.remove(nif);
                        System.out.println("Cliente con NIF " + nif + " eliminado con éxito.");
                    } else {
                        System.out.println("Cliente no encontrado.");
                    }
                    break;

                case "3":
                    // Mostrar los datos de un cliente
                    System.out.print("Ingrese el NIF del cliente: ");
                    nif = scanner.nextLine();
                    if (clientes.containsKey(nif)) {
                        System.out.println("Datos del cliente " + nif + ": " + clientes.get(nif));
                    } else {
                        System.out.println("Cliente no encontrado.");
                    }
                    break;

                case "4":
                    // Listar todos los clientes
                    if (!clientes.isEmpty()) {
                        System.out.println("Lista de todos los clientes:");
                        for (Map.Entry<String, Map<String, String>> entry : clientes.entrySet()) {
                            System.out.println("NIF: " + entry.getKey() + ", Nombre: " + entry.getValue().get("Nombre"));
                        }
                    } else {
                        System.out.println("No hay clientes en la base de datos.");
                    }
                    break;

                case "5":
                    // Listar clientes preferentes
                    boolean hayPreferentes = false;
                    for (Map.Entry<String, Map<String, String>> entry : clientes.entrySet()) {
                        if (entry.getValue().get("Preferente").equals("sí")) {
                            if (!hayPreferentes) {
                                System.out.println("Lista de clientes preferentes:");
                                hayPreferentes = true;
                            }
                            System.out.println("NIF: " + entry.getKey() + ", Nombre: " + entry.getValue().get("Nombre"));
                        }
                    }
                    if (!hayPreferentes) {
                        System.out.println("No hay clientes preferentes en la base de datos.");
                    }
                    break;

                case "6":
                    // Terminar el programa
                    System.out.println("Terminando el programa.");
                    scanner.close();
                    return;

                default:
                    System.out.println("Opción no válida. Intente nuevamente.");
                    break;
            }
        }
    }
}

File: python/Ejercicio_10.py
# Diccionario para almacenar clientes
clientes = {}

while True:
    # Mostrar el menú de opciones
    print("\nSeleccione una opción:")
    print("1. Añadir cliente")
    print("2. Eliminar cliente")
    print("3. Mostrar cliente")
    print("4. Listar todos los clientes")
    print("5. Listar clientes preferentes")
    print("6. Terminar")
    opcion = input("Ingrese su opción: ")

    if opcion == "1":
        # Añadir un nuevo cliente
        nif = input("Ingrese el NIF del cliente: ")
        nombre = input("Ingrese el nombre del cliente: ")
        direccion = input("Ingrese la dirección del cliente: ")
        telefono = input("Ingrese el teléfono del cliente: ")
        correo = input("Ingrese el correo electrónico del cliente: ")
        preferente = input("¿Es un cliente preferente? (sí/no): ").strip().lower() == "sí"

        clientes[nif] = {
            "Nombre": nombre,
            "Dirección": direccion,
            "Teléfono": telefono,
            "Correo": correo,
            "Preferente": preferente
        }
        print(f"Cliente {nombre} añadido con éxito.")

    elif opcion == "2":
        # Eliminar un cliente
        nif = input("Ingrese el NIF del cliente a eliminar: ")
        if clientes.pop(nif, None):
            print(f"Cliente con NIF {nif} eliminado con éxito.")
        else:
            print("Cliente no encontrado.")

    elif opcion == "3":
        # Mostrar datos de un cliente
        nif = input("Ingrese el NIF del cliente: ")
        cliente = clientes.get(nif)
        if cliente:
            print(f"Datos del cliente {nif}: {cliente}")
        else:
            print("Cliente no encontrado.")

    elif opcion == "4":
        # Listar todos los clientes
        if clientes:
            print("Lista de todos los clientes:")
            for nif, datos in clientes.items():
                print(f"NIF: {nif}, Nombre: {datos['Nombre']}")
        else:
            print("No hay clientes en la base de datos.")

    elif opcion == "5":
        # Listar clientes preferentes
        preferentes = {nif: datos for nif, datos in clientes.items() if datos["Preferente"]}
        if preferentes:
            print("Lista de clientes preferentes:")
            for nif, datos in preferentes.items():
                print(f"NIF: {nif}, Nombre: {datos['Nombre']}")
        else:
            print("No hay clientes preferentes en la base de datos.")

    elif opcion == "6":
        # Terminar el programa
        print("Terminando el programa.")
        break

    else:
        print("Opción no válida. Intente nuevamente.")
